src/allmydata/test/test_log.py

```diff
@@ -0,0 +1,113 @@
+"""
+Tests for allmydata.util.log.
+
+Ported to Python 3.
+"""
+
+from __future__ import unicode_literals
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from future.utils import PY2
+if PY2:
+    from builtins import filter, map, zip, ascii, chr, hex, input, next, oct, open, pow, round, super, bytes, dict, int, list, object, range, str, max, min  # noqa: F401
+
+from twisted.trial import unittest
+from twisted.python.failure import Failure
+
+from foolscap.logging import log
+
+from allmydata.util import log as tahoe_log
+
+
+class SampleError(Exception):
+    pass
+
+
+class LoggingObject0(tahoe_log.PrefixingLogMixin):
+    pass
+
+
+class LoggingObject1(tahoe_log.PrefixingLogMixin):
+    pass
+
+
+class LoggingObject2(tahoe_log.PrefixingLogMixin):
+    pass
+
+
+class LoggingObject3(tahoe_log.PrefixingLogMixin):
+    pass
+
+
+class Log(unittest.TestCase):
+    def setUp(self):
+        self.messages = []
+
+        def msg(msg, facility, parent, *args, **kwargs):
+            self.messages.append((msg, facility, parent, args, kwargs))
+            return "msg{}".format(len(self.messages))
+
+        self.patch(log, "msg", msg)
+
+    def test_err(self):
+        """Logging with log.err() causes tests to fail."""
+        try:
+            raise SampleError("simple sample")
+        except:
+            f = Failure()
+        tahoe_log.err(format="intentional sample error",
+                      failure=f, level=tahoe_log.OPERATIONAL, umid="wO9UoQ")
+        self.flushLoggedErrors(SampleError)
```

Comment:
I'd probably add an assertion here, that `flushLoggedErrors` returned a list with one element, and change the docstring to just say that `log.err` forwards errors to the Twisted log (and then it's up to trial to make the test fail, not our concern).

Reply:
Oh.  I see.  This test was moved from the other module. :/  Where it was also testing nothing...  Well, the changes would still be good, I guess, and they're basically trivial.